### DISCUSSION TITLE
Changing Account - What Happens?

### DISCUSSION THREAD

**Discussion by ghost** [tsightler/ring-mqtt#675] [Support Q&A] [answered]
Hello,

If I decide to force re-authentication and use a different account that is set up with the same levels of access and such to my ring devices, will this affect any of my data/automation/entities etc... within HomeAssistant? Or will it just see that it is the same "Location" or whatever you want to call it that all my ring devices are tied to and then see that all the devices already exist in HA and essentially just authenticate and nothing else?

Ideally I would like to change from my account to one dedicated to my instance of HomeAssistant but without causing any issues/conflicts/additional work for me relating to my HA instance.

Thanks!

**Comment by tsightler** [comment, marked as answer]
All device ids as well as topics levels are generated from the location/device IDs provided by the Ring API and are not associated with the account.  Switching accounts should have no significant impact other than the fact that secondary accounts can't access certain minor features/devices, for example they cannot change the volume of the base station and cannot access chimes devices.

I switch between my primary and secondary accounts on my dev instance for testing with no changes to devices all the time.  The entire design goal of ring-mqtt is that topics are consistent.  Of course I can't be sure it will cause you no problems, but I wouldn't expect any.  Worst case you can take a backup and then revert, or just swtich back to the original account.

**Comment by ghost** [comment]
Awesome - thank you so much for the reply, and all you do with this project! :)

**Comment by ghost** [comment]
Unfortunately I ended up doing this and I wasn’t receiving the expected results (i.e. my Ring app notifications were still returning that the alarm arm/disarm was triggered by the old account that was logged in, rather than the new account. Forcing a sign out of all devices from the ring app and then reauthenticating with my second account causes issues with third party integrations such as water line, garage door, deadbolt, etc…)

I have signed back into my old account now and am wondering if there is a way to implement the “disarm” code to actually tell Ring that it was disarmed by the users who’s code was sent - i.e. user my secondary code to reflect the user (i have a code for a user named “Home Assistant”). Even in my automations and such when it asks for a disarm code I provide the one for the “Home Assistant” user i have configured in my Ring environment, but it still shows it being disabled /enabled by my user instead.

thank you!

**Reply by tsightler** [reply]
The safest way to do it is to stop ring-mqtt, delete all ring-mqtt authenticated users from Ring Control Center, then start ring-mqtt and re-authenticate with the new user account.  This will force only the new refresh token associated with the new account to be valid.  If you just re-authenticate while everything is otherwise running then it might not switch users on the fly.

> am wondering if there is a way to implement the “disarm” code to actually tell Ring that it was disarmed by the users who’s code was sent

No, it's not.

**Reply by tsightler** [reply]
Sorry, last answer was a little abrupt as I was answering on my phone.  It's not possible for ring-mqtt to tell Ring what user is disarming because that's not how the API for the app works, and ring-mqtt can only do what the app can do.  Think about it this way, in the Ring app, can you tell it who is arming/disarming?  No, it's only based on who is logged into the app.  There's no difference with ring-mqtt, basically, ring-mqtt is just another app.

The Ring keypad works completely differently.  It connects to the Ring base station via Zwave and sends button presses to the base station.  When you create codes in the app, those codes are pushed to the base station via the websocket connection so the base station can validate the codes entered and identify users based on them.

Now, I'm not sure what exactly you are trying to accomplish, or why you need Ring to know who is arming/disarming via HA, but you can use a panel like the Alarmo panel and rules to sync it with the Ring panel, and then HA can know who is arming/disarming.

**Reply by ghost** [reply]
This worked like a charm, thanks so much for the reply and all you do with this project!

To satisfy any curiosity, I basically just wanted to have notifications directly from Ring (and likewise in the Ring logs) show that it was disarmed/armed by "Home Assistant" rather than by my user account. An odd little request, but just something I prefer to have reflected from the Ring side as my wife prefers Ring notifications for some things instead of HomeAssistant (my guess is as good as yours).
